packages/flutter_tools/lib/src/compile.dart

```diff
@@ -16,11 +16,24 @@ import 'base/io.dart';
 import 'base/platform.dart';
 import 'base/process_manager.dart';
 import 'base/terminal.dart';
+import 'codegen.dart';
 import 'convert.dart';
 import 'dart/package_map.dart';
 import 'globals.dart';
+import 'project.dart';
 
-KernelCompiler get kernelCompiler => context[KernelCompiler];
+KernelCompilerFactory get kernelCompilerFactory => context[KernelCompilerFactory];
+
+class KernelCompilerFactory {
+  const KernelCompilerFactory();
+
+  Future<KernelCompiler> create(FlutterProject flutterProject) async {
+    if (flutterProject == null || !flutterProject.hasBuilders) {
```

Comment:
Is it possible for a project to go from not having builders to having some while a `flutter` command is running? If so, is this case handled? IOW, would the factory be rerun to get the code-generating compiler?

Reply:
Yes,  we won't be able to switch them currently. This does fall under the umbrella for "hot reload literally everything" that I am working on.

The solution is to: 1) always run with build_runner enabled and 2) use manifest invalidation as triggers to update the build script

However, that is a much bigger change. I don't think it should block letting people use the feature

Reply:
Makes sense.